docs/source/system-architecture.md

```diff
@@ -112,6 +112,14 @@ within a Docker Swarm cluster.
 within Docker configuration.  Note: because these kernels will always run local to the corresponding Enterprise Gateway instance, these process proxies are of limited use.
 - `ConductorClusterProcessProxy` - is responsible for the discovery and management of kernels hosted
 within an IBM Spectrum Conductor cluster.
+- `SparkOperatorProcessProxy` - is responsible for the descovery and management of kernels hosted 
+within a Kubernetes cluster but created as a SparkApplication instead of a Pod. The SparkApplication is a Kubernetes custom resource 
+defined inside the project [spark-on-k8s-operator](https://github.com/GoogleCloudPlatform/spark-on-k8s-operator), which 
+makes all kinds of spark on k8s components better organized and easy to configure.
+
+Before you run this kernel, please ensure that spark operator is installed under a certain namespace of your Kubernetes
```

Comment:
I'm not sure I understand this statement: _ensure that spark operator is installed under a certain namespace_.

Does the `SparkOperatorProcessProxy` need to know _that_ namespace or are you saying that the spark operator just needs to be deployed with a known namespace?  How is this different from today's kernel pods?

Reply:
yep, the spark operator includes not only crd(SparkApplication), but also an operator pod which intend to inspect the running status of all running application as well as hook some action when events(start, end, error) happened to any application.

the pod must be deployed under a certain namespace first for start a spark application, so you understanding is correct.